Keas.Mvc/ClientApp/components/Keys/KeySerialTable.tsx

```diff
@@ -0,0 +1,209 @@
+                    {
+                        Filter: ({ filter, onChange }) => (
+                            <select
+                                onChange={e => onChange(e.target.value)}
+                                style={{ width: "100%" }}
+                                value={filter ? filter.value : "all"}
+                            >
+                                <option value="all">Show All</option>
+                                <option value="active">Active</option>
+                                <option value="inactive">Inactive</option>
+                            </select>
+                        ),
+                        Header: "Status",
+                        accessor: "status",
+                        filterMethod: (filter: IFilter, row: IKeySerial) => {
+                            if (filter.value === "all") {
+                                return true;
+                            }
+                            if (filter.value === "active") {
+                                return row.status === "Active";
+                            }
+                            if (filter.value === "inactive") {
+                                return row.status !== "Active";
+                            }
+                        }
+                    },
+                    {
+                        Header: "Assignment",
+                        accessor: (keySerial: IKeySerial) =>
+                            keySerial.keySerialAssignment
+                                ? keySerial.keySerialAssignment.person.name
+                                : null,
+                        className: "word-wrap",
+                        filterMethod: (filter: IFilter, row: IRow) =>
+                            !!row[filter.id] &&
+                            row[filter.id].toLowerCase().includes(filter.value.toLowerCase()),
+                        id: "assignedTo"
+                    },
+                    {
+                        Cell: row => (
+                            <span>{row.value ? DateUtil.formatExpiration(row.value) : ""}</span>
+                        ),
+                        Filter: ({ filter, onChange }) => (
+                            <select
+                                onChange={e => onChange(e.target.value)}
+                                style={{ width: "100%" }}
+                                value={filter ? filter.value : "all"}
+                            >
+                                <option value="all">Show All</option>
+                                <option value="unassigned">Unassigned</option>
+                                <option value="expired">Expired</option>
+                                <option value="unexpired">All Unexpired</option>
+                                <option value="3weeks">Expiring within 3 weeks</option>
+                                <option value="6weeks">Expiring within 6 weeks</option>
+                            </select>
```

Comment:
We use this same pattern quite a bit, including repeating these exact options.  Next time we change a component let's extract to a new shared expiration filter component.